### PR TITLE
fix: incessant requerying of nfts with missing metadata

### LINKED ISSUE #5716: Prevent reload of data for Stacks NFTs without metadata

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out Leather build 4b54db2 — [Extension build](https://github.com/leather-io/extension/actions/runs/10317750417), [Test report](https://leather-io.github.io/playwright-reports/fix-dead-nft-queries), [Storybook](https://fix-dead-nft-queries--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=fix-dead-nft-queries)

This PR adds some state to prevent requerying of metadata when we know that it isn't, and won't, be there in future. This'll save _loads_ of requests to Hiro's APIs.

This PR only tackles NFTs. Perhaps you can follow up doing the same with FTs @alter-eggo?

Adding this highlights some of the failures of the query package, I think. For example, the "query hook" in question was actually a wrapper hook, preventing accessing the internals. A rule for query package should be that **ONLY** options creator functions are returned. I think we shouldn't even return wrapped `useQueries` as it get confusing when we need to create an app-specific clone of same name.

**Before**

https://github.com/user-attachments/assets/b74e5818-115f-4605-ae4b-f37c4d8cf026

**After**
_Now only has FT missing queries_

https://github.com/user-attachments/assets/0454ebe7-6b03-4697-a978-3922b41e607a

### CODE REVIEW

File: src/app/features/collectibles/components/stacks/stacks-crypto-assets.tsx [hunk 1]
```diff
@@ -18,7 +16,13 @@ interface StacksCryptoAssetsProps {
 export function StacksCryptoAssets({ address }: StacksCryptoAssetsProps) {
   const names = useGetBnsNamesOwnedByAddressQuery(address).data?.names;
 
-  const stacksNftsMetadataResp = useStacksNonFungibleTokensMetadata(address);
+  // NftMetadataResponse[]
```
Comment (alter-eggo) [inline src/app/features/collectibles/components/stacks/stacks-crypto-assets.tsx:19]: this should be removed I believe?

**Review by alter-eggo — APPROVED** [review]

**Comment by markmhendrickson** [comment]
@alter-eggo to [implement similar solution for FTs](https://github.com/leather-io/extension/issues/5755) to compare with this PR

**Comment by pete-watters** [comment]
@alter-eggo do you think this is OK to merge and get released? Once we remove [this code ](https://github.com/leather-io/extension/pull/5747/files#r1711363030)

**Comment by alter-eggo** [comment]
@pete-watters I think we can close this pr

**Comment by pete-watters** [comment]
> @pete-watters I think we can close this pr

We can merge it? Or just close it as the work is no longer needed?

**Comment by alter-eggo** [comment]
@pete-watters just closed, problem is resolved here https://github.com/leather-io/mono/pull/377

**Pull request closed without merging**